packages/react-components/react-toast/src/components/ToastBody/useToastBodyStyles.styles.ts

```diff
@@ -27,18 +28,34 @@ const useSubtitleBaseClassName = makeResetStyles({
   color: tokens.colorNeutralForeground2,
 });
 
+const useInvertedStyles = makeStyles({
+  root: {
+    color: tokens.colorNeutralForegroundInverted2,
+  },
+  subtitle: {
+    color: tokens.colorNeutralForegroundInverted2,
+  },
+});
+
 /**
  * Apply styling to the ToastBody slots based on the state
  */
 export const useToastBodyStyles_unstable = (state: ToastBodyState): ToastBodyState => {
   const rootBaseClassName = useRootBaseClassName();
   const subtitleBaseClassName = useSubtitleBaseClassName();
-  state.root.className = mergeClasses(toastBodyClassNames.root, rootBaseClassName, state.root.className);
+  const invertedStyles = useInvertedStyles();
+  state.root.className = mergeClasses(
+    toastBodyClassNames.root,
+    rootBaseClassName,
+    state.appearance === 'inverted' && invertedStyles.root,
+    state.root.className,
+  );
 
   if (state.subtitle) {
     state.subtitle.className = mergeClasses(
       toastBodyClassNames.subtitle,
       subtitleBaseClassName,
+      state.appearance === 'inverted' && invertedStyles.subtitle,
```

Comment:
nit: you could just invoke `useToastContext` directly here instead of adding it to the state of `Toastbody`

Reply:
That breaks our hook encapsulation pattern 🤔